addons/account/models/sequence_mixin.py

```diff
@@ -45,20 +45,49 @@ def _check_grouping(grouping, optional=None, required=None):
             return 'never'
         raise ValidationError(_('The sequence regex should at least contain the prefix1, seq and suffix grouping keys. For instance:\n^(?P<prefix1>.*?)(?P<seq>\d*)(?P<suffix>\D*?)$'))
 
+    def _constrains_regex_match(self):
+        try:
+            for rec in self:
+                name = rec[rec._sequence_field]
+                rec._deduce_sequence_number_reset(name)
+        except ValidationError:
+            raise ValidationError(_("Field %s should have a numeric part.") % (rec._sequence_field,)) from None
+
+    def __init__(self, pool, cr):
+        api.constrains(self._sequence_field)(type(self)._constrains_regex_match)
+        return super().__init__(pool, cr)
+
+    def _reset_mode_to_order(self, mode):
+        """Generate an `ORDER BY` statement from a sequence matching regex"""
+        if not mode:
+            return f"{self._sequence_field} DESC"
+
+        regex = {
+            "month": self._sequence_monthly_regex,
+            "year": self._sequence_yearly_regex,
+            "never": self._sequence_fixed_regex,
+        }[mode]
+
+        regex = regex.replace(r"?P<seq>", "")
+        # convert other groups to non-capturing groups.
+        regex = re.sub(r"\?P<\w+>", "?:", regex)
+        return self.env.cr.mogrify(f"(regexp_match({self._sequence_field}, %s))[1]::integer DESC", [regex]).decode()
```

Comment:
Couldn't you use a 
```sql
('0' || regexp_match({self._sequence_field}, %s))[1])::integer DESC
````
Instead of adding a constraint on the regex used  by forcing the `+` in  `(?P<seq>\d+)`, and by adding a hacky constraint in the `__init__`?

Reply:
Maybe.